frontend/src/app/tasks/components/EditTask.tsx

```diff
@@ -0,0 +1,115 @@
+import Box from '@mui/joy/Box'
+import Stack from '@mui/joy/Stack'
+
+import { Task } from '@/domain/Task'
+import { TaskType } from '@/domain/TaskType'
+import { Project } from '@/domain/Project'
+import Button from '@mui/joy/Button'
+
+import { Select } from '@/ui/Select/Select'
+import { TextArea } from '@/ui/TextArea/TextArea'
+import { Input } from '@/ui/Input/Input'
+
+import { TimePicker } from './TimePicker'
+import { useEditTaskForm } from '../hooks/useEditTaskForm'
+
+type EditTaskProps = {
+  task: Task
+  tasks: Array<Task>
+  projects: Array<Project>
+  taskTypes: Array<TaskType>
+  closeForm: () => void
+}
+
+export const EditTask = ({ task, tasks, projects, taskTypes, closeForm }: EditTaskProps) => {
+  const { handleChange, formState, handleSubmit, resetForm } = useEditTaskForm({
```

Comment:
This is also a question in order to learn better the architecture, why do we need hooks and not simple functions for the CRUD methods?

Reply:
And also why do we have the useEditTaskForm and useEditTask? Sometimes it feels we have too many layers or things are starting to get too over engineered but it could be lack of knowledge from my side.

Reply:
ok thanks!